node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeList.java

```diff
@@ -91,6 +91,11 @@ public NodeList changingVersion() {
                                 !node.status().vespaVersion().get().equals(node.allocation().get().membership().cluster().vespaVersion()));
     }
 
+    /** Returns the subset of nodes with want to fail status matching the given value */
+    public NodeList wantToFail(boolean wantToFail) {
```

Comment:
This should not take a `boolean` as the parent class of this provides inverting through `not()`. E.g. you can do `list.wantToFail()` and `list.not().wantToFail()` to invert the condition.

Consider naming this method `failing`, similar to the existing `deprovisioning` and `rebuilding` methods.

Reply:
Yes, the only drawback with not passing a boolean is the cases where it is a parameter in the caller. Anyway, I'll follow the pattern, I didn't find one, but failing etc. is fine.